common/client/node_lifecycle_test.go

```diff
@@ -496,6 +494,123 @@ func TestUnit_NodeLifecycle_aliveLoop(t *testing.T) {
+		t.Parallel()
+		rpc := newMockNodeClient[types.ID, Head](t)
+		rpc.On("GetInterceptedChainInfo").Return(ChainInfo{}, ChainInfo{}).Once()
+		ch := make(chan Head)
+		close(ch)
+		rpc.On("SubscribeToFinalizedHeads", mock.Anything).Return((<-chan Head)(ch), newSub(t), nil).Once()
+		lggr, observedLogs := logger.TestObserved(t, zap.DebugLevel)
+		node := newSubscribedNode(t, testNodeOpts{
+			chainConfig: clientMocks.ChainConfig{
+				IsFinalityTagEnabled: true,
+			},
+			rpc:  rpc,
+			lggr: lggr,
+		})
+		defer func() { assert.NoError(t, node.close()) }()
+		// disconnects all on transfer to unreachable or outOfSync
+		rpc.On("DisconnectAll").Once()
+		// might be called in unreachable loop
+		rpc.On("Dial", mock.Anything).Return(errors.New("failed to dial")).Maybe()
+		node.declareAlive()
+		tests.AssertLogEventually(t, observedLogs, "Finalized heads subscription channel unexpectedly closed")
+		tests.AssertEventually(t, func() bool {
+			return nodeStateUnreachable == node.State()
+		})
+	})
+	t.Run("when no new finalized heads received for threshold, transitions to out of sync", func(t *testing.T) {
+		t.Parallel()
+		rpc := newMockNodeClient[types.ID, Head](t)
+		rpc.On("GetInterceptedChainInfo").Return(ChainInfo{}, ChainInfo{}).Once()
+		ch := make(chan Head, 1)
+		ch <- head{BlockNumber: 10}.ToMockHead(t)
+		rpc.On("SubscribeToFinalizedHeads", mock.Anything).Return((<-chan Head)(ch), newSub(t), nil).Once()
+		lggr, observed := logger.TestObserved(t, zap.DebugLevel)
+		noNewFinalizedHeadsThreshold := tests.TestInterval
+		node := newSubscribedNode(t, testNodeOpts{
+			config: testNodeConfig{},
+			chainConfig: clientMocks.ChainConfig{
+				NoNewFinalizedHeadsThresholdVal: noNewFinalizedHeadsThreshold,
+				IsFinalityTagEnabled:            true,
+			},
+			rpc:  rpc,
+			lggr: lggr,
+		})
+		defer func() { assert.NoError(t, node.close()) }()
+		// tries to redial in outOfSync
+		rpc.On("Dial", mock.Anything).Return(errors.New("failed to dial")).Run(func(_ mock.Arguments) {
+			assert.Equal(t, nodeStateOutOfSync, node.State())
+		}).Once()
+		// disconnects all on transfer to unreachable or outOfSync
+		rpc.On("DisconnectAll").Maybe()
+		// might be called in unreachable loop
+		rpc.On("Dial", mock.Anything).Return(errors.New("failed to dial")).Maybe()
+		node.declareAlive()
+		tests.AssertLogEventually(t, observed, fmt.Sprintf("RPC's finalized state is out of sync; no new finalized heads received for %s (last finalized head received was 10)", noNewFinalizedHeadsThreshold))
+		tests.AssertEventually(t, func() bool {
+			// right after outOfSync we'll transfer to unreachable due to returned error on Dial
+			// we check that we were in out of sync state on first Dial call
+			return node.State() == nodeStateUnreachable
+		})
+	})
+	t.Run("when no new heads received for threshold but we are the last live node, forcibly stays alive", func(t *testing.T) {
```

Comment:
nit: to avoid confusion with the similar test above, think we should mention this tests specifically no new finalized head
```suggestion
	t.Run("when no new finalized heads received for threshold but we are the last live node, forcibly stays alive", func(t *testing.T) {
```